controller/src/main/java/io/pravega/controller/store/stream/tables/TableHelper.java

```diff
@@ -645,4 +645,61 @@ public static boolean isRerunOf(final List<Integer> segmentsToSeal,
 
         return historyRecordOpt;
     }
+
+    public static boolean isScaleInputValid(final List<Integer> segmentsToSeal,
+                                            final List<AbstractMap.SimpleEntry<Double, Double>> newRanges,
+                                            final byte[] segmentTable) {
+        boolean newRangesPredicate = newRanges.stream().noneMatch(x -> x.getKey() >= x.getValue() &&
+                x.getKey() >= 0 && x.getValue() > 0);
+
+        List<AbstractMap.SimpleEntry<Double, Double>> oldRanges = segmentsToSeal.stream()
+                .map(segment -> SegmentRecord.readRecord(segmentTable, segment).map(x ->
+                        new AbstractMap.SimpleEntry<>(x.getRoutingKeyStart(), x.getRoutingKeyEnd())))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .collect(Collectors.toList());
+
+        return newRangesPredicate && reduce(oldRanges).equals(reduce(newRanges));
+    }
+
+    /**
+     * Helper method to compute list of continuous ranges. For example, two neighbouring key ranges where,
+     * range1.high == range2.low then they are considered neighbours.
+     * This method reduces input range into distinct continuous blocks.
+     * @param input list of key ranges.
+     * @return reduced list of key ranges.
+     */
+    private static List<AbstractMap.SimpleEntry<Double, Double>> reduce(List<AbstractMap.SimpleEntry<Double, Double>> input) {
+        List<AbstractMap.SimpleEntry<Double, Double>> ranges = new ArrayList<>(input);
+        ranges.sort(Comparator.comparingDouble(AbstractMap.SimpleEntry::getKey));
+        List<AbstractMap.SimpleEntry<Double, Double>> result = new ArrayList<>();
+        double low = -1.0;
+        double high = -1.0;
+        for (AbstractMap.SimpleEntry<Double, Double> range : ranges) {
+            if (high < range.getKey()) {
+                // add previous result and start a new result if prev.high is less than next.low
+                if (low != -1.0 && high != -1.0) {
+                    result.add(new AbstractMap.SimpleEntry<>(low, high));
+                }
+                low = range.getKey();
+                high = range.getValue();
+            } else if (high == range.getKey()) {
+                // if adjacent (prev.high == next.low) then update only high
+                high = range.getValue();
+            } else {
```

Comment:
> range.getKey() < low

is not possible because its sorted by range.key

> range encompasses [low, high)
this is only possible if range.low == low.
Now since high is strictly greater than low, it implies high > range.low. That case is covered 


I can add more test cases to ensure that everything is covered